File: api/src/main/java/com/mindcare/api/model/Notification.java
package com.mindcare.api.model;

import lombok.Builder;
import lombok.Data;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Data
@Builder
public class Notification {
  
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String userType;

  @Column(nullable = false)
  private String value;

  @Column(name = "created_at", nullable = false)
  @Builder.Default
  private LocalDateTime createdAt = LocalDateTime.now();

  @Column(nullable = true)
  private LocalDateTime deletedAt;

  @ManyToOne
  @JoinColumn(name = "user_uuid", referencedColumnName = "uuid")
  private User user;
}

File: api/app/Models/Notification.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Notification extends Model
{
  protected $table = 'notifications';

  protected $fillable = [
    'user_type',
    'value',
    'user_uuid'
  ];

  protected $casts = [
    'created_at' => 'datetime',
    'deleted_at' => 'datetime',
  ];

  use SoftDeletes;

  public function user(): BelongsTo
  {
    return $this->belongsTo(User::class, 'user_uuid', 'uuid');
  }
}
